### ISSUE TITLE
Error installing Net::Stripe

### ISSUE THREAD

**Issue by abh** [lukec/stripe-perl#202]
```
"my" variable $h masks earlier declaration in same scope at /usr/local/share/perl5/site_perl/HTTP/Headers.pm line 184.

#   Failed test 'use Net::Stripe;'
#   at t/local.t line 12.
#     Tried to use 'Net::Stripe'.
#     Error:  Can't call method "add_attribute" on an undefined value at /usr/local/share/perl5/site_perl/Moo/HandleMoose.pm line 169.
```

to reproduce:

```
docker run --name net-stripe-test -ti alpine:3.11
# the rest in the docker container
apk update
apk upgrade
apk add perl-app-cpanminus perl-dev build-base wget gcc make openssl-dev openssl curl git bash
cpanm Net::Stripe
# it fails, see test logs and then
cpanm -n Net::Stripe
```

 but the tests were right, something isn't working right:

```
/ # perl -MNet::Stripe -e ''
"my" variable $h masks earlier declaration in same scope at /usr/local/share/perl5/site_perl/HTTP/Headers.pm line 184.
Can't call method "add_attribute" on an undefined value at /usr/local/share/perl5/site_perl/Moo/HandleMoose.pm line 169.
Compilation failed in require at /usr/local/share/perl5/site_perl/Net/Stripe.pm line 28.
BEGIN failed--compilation aborted at /usr/local/share/perl5/site_perl/Net/Stripe.pm line 28.
Compilation failed in require.
BEGIN failed--compilation aborted.
```

I don't get that error from HTTP::Headers by just loading that alone.

**Comment by abh** [comment]
FWIW, the add_attribute error goes away if I comment out `with Throwable`.

**Comment by sherrardb** [comment]
@abh, thanks for the report. can you confirm whether this is related to the same `Devel::CallParser` bug from #114?
[https://github.com/tobyink/p5-kavorka/issues/22](https://github.com/tobyink/p5-kavorka/issues/22)
[https://rt.cpan.org/Public/Bug/Display.html?id=109841](https://rt.cpan.org/Public/Bug/Display.html?id=109841)

it seems like i have seen that same message in the cpantesters output for the same issue.

**Comment by sherrardb** [comment]
BTW, what did you end up doing about it in #114?

**Comment by abh** [comment]
It seems different than #114.  I went back to Go, but I guess I forgot and thought I should give the Perl version another try!

Kavorka installed and tested okay, so I think this is a separate issue.

**Comment by sherrardb** [comment]
currently installing threaded perl 5.22 so that i can confirm, my searches on that particular message lead me back down a similar trail as before. difference route, same final destination.

https://github.com/maxmind/App-GHPT/issues/7#issuecomment-300975231
https://github.com/maxmind/App-GHPT/issues/7#issuecomment-300975310
https://rt.cpan.org/Public/Bug/Display.html?id=121607

**Comment by abh** [comment]
Is it this one? It's from 2015, so ... sad if so.  https://rt.cpan.org/Public/Bug/Display.html?id=110623

**Comment by sherrardb** [comment]
that is the one, and i concur. but such is the reality of community-maintained software. and it's not like commercial software doesn't have its share of warts. :-)

still waiting on my perlbrew install. i keep forgetting about the Y2020 bug in Time::Local on older perls.

**Comment by sherrardb** [comment]
@abh, can you confirm the version of `Net::Stripe` that being installed in your container? i am (finally) getting the same series of errors, but mine are triggered on a distinctly different line in `Net::Stripe`.

**Comment by sherrardb** [comment]
and can you confirm the output of the following tests:

```
~$ perl -mLWP::UserAgent -e ''
~$ 
~$ perl -mDevel::CallParser -mLWP::UserAgent -e ''
"my" variable $h masks earlier declaration in same scope at /home/sburton/perl5/perlbrew/perls/perl-5.22.4/lib/site_perl/5.22.4/HTTP/Headers.pm line 184.
```

```
~$ perl -mMoose -e '{ package Dummy; use Moose; with("Throwable"); 1;}'
~$ 
~$ perl -mDevel::CallParser -mMoose -e '{ package Dummy; use Moose; with("Throwable"); 1;}'
"my" variable $meta masks earlier declaration in same scope at /home/sburton/perl5/perlbrew/perls/perl-5.22.4/lib/site_perl/5.22.4/x86_64-linux-thread-multi/Class/MOP/Class.pm line 92.
"my" variable $method masks earlier declaration in same scope at /home/sburton/perl5/perlbrew/perls/perl-5.22.4/lib/site_perl/5.22.4/x86_64-linux-thread-multi/Moose/Meta/Attribute.pm line 1166.
"my" variable $class_method masks earlier declaration in same scope at /home/sburton/perl5/perlbrew/perls/perl-5.22.4/lib/site_perl/5.22.4/x86_64-linux-thread-multi/Moose/Meta/Role/Application/ToClass.pm line 157.
"my" variable $role2_method masks earlier declaration in same scope at /home/sburton/perl5/perlbrew/perls/perl-5.22.4/lib/site_perl/5.22.4/x86_64-linux-thread-multi/Moose/Meta/Role/Application/ToRole.pm line 105.
Can't call method "add_attribute" on an undefined value at /home/sburton/perl5/perlbrew/perls/perl-5.22.4/lib/site_perl/5.22.4/Moo/HandleMoose.pm line 169.
```

**Comment by abh** [comment]
Net::Stripe 0.42

And yes, I see the same errors (and lack of errors) as you do (same line numbers, too).

Thank you for looking into this; I realize all of this is pretty far from Net::Stripe itself (other than having chosen the dependencies).

**Comment by sherrardb** [comment]
@abh, no worries. i had to field a similar question recently for someone who was forced to do a fresh OS install. i pointed him to the same threads, and even though i did not have time to decipher exactly what he patched, his final response was:
"installed the patch mentioned in https://github.com/tobyink/p5-kavorka/issues/22 and am able to install Net::Stripe."

**Comment by abh** [comment]
yeah, indeed -- installing Alt::Devel::CallParser::ButWorking seems to fix my test setup. I'll try it in the "real" docker container too now.

**Comment by abh** [comment]
Yes, that fixed it. Thank you for the help!

I'm too out of touch with the Perl best practices to suggest if you should make Alt::Devel::CallParser::ButWorking a dependency, but it'd have worked for me.

**Comment by sherrardb** [comment]
given the scope of packages impacted by this bug,i am hoping that someone will take up the mantle of `Devel::CallParser` and we won't have to go down that road.

we'll see.

**Comment by abh** [comment]
Apparently not yet! :-(  (And now Alt::Devel::CallParser::ButWorking doesn't install cleanly either).

**Comment by williamrhaynes** [comment]
cannot install Net::Stripe on Ubuntu 22.04...Also unable to install Alt::Devel::CallParser::ButWorking
can anyone help?
bill
